src/main/kotlin/com/kotlindiscord/bot/extensions/CleanExtension.kt

```diff
@@ -0,0 +1,162 @@
+
+                    for (channelId in channels) {
+                        var query = bot.kord.cache.query<MessageData> {
+                            MessageData::channelId eq channelId
+
+                            if (!userIds.isNullOrEmpty()) run {
+                                MessageData::authorId `in` userIds
+                            }
+
+                            if (botonly) run {
+                                MessageData::authorIsBot eq true
+                            }
+
+                            if (!regex.isNullOrEmpty()) run {
+                                MessageData::content predicate { regex.all { regex -> regex.matches(it) } }
+                            }
+
+                            if (sinceTimestamp != null) run {
+                                MessageData::timestamp predicate {
+                                    Instant.parse(it).isAfter(sinceTimestamp)
+                                }
+                            }
+                        }.toCollection()
+
+                        if (count > 0) {
+                            query = query.sortedBy { Instant.parse(it.timestamp) }.reversed().slice(0..count)
+                        }
+
+                        if (query.size > MAX_DELETION_SIZE) {
+                            if (
+                                message.getAuthorAsMember()?.roles?.toList()
+                                    ?.contains(config.getRole(Roles.ADMIN)) == false
+                                // Basically, if the author isn't an admin.
+                            ) {
+                                message.channel.createMessage(
+                                    ":x: Cannot delete more than $MAX_DELETION_SIZE, " +
+                                            "please ask an admin to run this command with the `force:true` flag."
+                                )
+                                return@action
+                            } else {
+                                if (!force) {
+                                    message.channel.createMessage(
+                                    ":x: Cannot delete more than $MAX_DELETION_SIZE, " +
+                                            "run this command with the `force:true` flag to force it."
+                                    )
+                                    return@action
+                                }
+                            }
+                        }
+
+                        val cleanCount = "Messages to clean: ${query.joinToString(", ") { it.id.toString() }}"
+                        logger.debug { cleanCount }
+
+                        val channel = bot.kord.getChannel(Snowflake(channelId))
+                        if (channel is GuildMessageChannel) {
+                            channel.bulkDelete(query.map { Snowflake(it.id) })
+                        } else {
+                            logger.warn { "Error retrieving channel $channelId : $channel" }
```

Comment:
You've got a space before a colon here. This is probably better formatted as `$channel ($channelId)` though.

Reply:
`channel` is an actual object, so we can see if it returned null, a DM channel, or whatever, it isn't a string, don't think that would make sense.

Reply:
I've been yelling at the Kord team about better toString methods, maybe we should make it an issue

Reply:
What do you think we should do then?

Reply:
I mean, there's nothing we _can_ do for now.